src/concrete/ml/torch/compile.py

```diff
@@ -81,10 +81,7 @@ def build_quantized_module(
         convert_torch_tensor_or_numpy_array_to_numpy_array(val) for val in to_tuple(torch_inputset)
     )
 
-    # Tracing needs to be done with the batch size of 1 since we compile our models to FHE with
-    # this batch size. The input set contains many examples, to determine a representative
-    # bit-width, but for tracing we only take a single one. We need the ONNX tracing batch size to
-    # match the batch size during FHE inference which can only be 1 for the moment.
+    # No batch dimension (i.e., 0 instead of [0]) because else GEMM onnx pass can't be applied
```

Comment:
is the gemm pass issue the only reason we have this ? looking at the above comments I'm not so sure